### ISSUE TITLE
PublicMethodsHaveTypeChecker should not flag defs defined inside val initializer blocks

### ISSUE THREAD

**Issue by leifwickland** [scalastyle/scalastyle#129]
This is related to #48. It is being created as a separate issue at the request of @matthewfarwell.

If a `val` receives its value from an anonymous block and a `def` is defined inside that block then scalastyle still treats it as a public method.

For example, in the following fancyStuff is detected as violating PublicMethodsHaveTypeChecker:

```scala
private val complexInit: Seq[Int] = {
  def fancyStuff(x: Int) = x * 2
  1.to(10).map(fancyStuff).toSeq
}
```
